src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -10,27 +10,35 @@ public partial class MemoryMappedFile
     {
         /// <summary>
         /// Used by the 2 Create factory method groups.  A null fileHandle specifies that the 
-        /// memory mapped file should not be associated with an exsiting file on disk (ie start
+        /// memory mapped file should not be associated with an existing file on disk (ie start
         /// out empty).
         /// </summary>
         [SecurityCritical]
-        private static SafeMemoryMappedFileHandle CreateCore(
-            SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability,
-            MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity)
+        private static unsafe SafeMemoryMappedFileHandle CreateCore(
+            SafeFileHandle fileHandle, String mapName, 
+            HandleInheritability inheritability, MemoryMappedFileAccess access, 
+            MemoryMappedFileOptions options, Int64 capacity)
         {
-            throw NotImplemented.ByDesign; // TODO: Implement this
+            if (mapName != null)
+            {
+                throw GetNamedMapsNotSupportedException();
```

Comment:
Super Nit: The "Get" prefix of this method makes me think it's returning some fixed data instead of throwing an exception.